### ISSUE TITLE
[2.2.0-RC 2.1] CMS widget data is not migrated to JSON data 

### ISSUE THREAD

**Issue by toddwolaver** [magento/magento2#10665]
### Description
When upgrading from 2.1.8 to 2.2.0-RC2.1 all serialized data is transformed to JSON data, except for widget data within CMS content.

If widget data within CMS content will not be migrated via the system upgrade, then this should be called out in documentation with a manual process identified.

### Preconditions
1. Install 2.1.8 via composer metapackage
2. Install sample data

### Steps to reproduce
1. Add a Product List widget to the top of the 'contact-us-info' CMS block. This should result in the addition of a widget such as this:
    1.  ```{{widget type="Magento\CatalogWidget\Block\Product\ProductsList" show_pager="0" products_count="10" template="product/widget/content/grid.phtml" conditions_encoded="a:2:[i:1;a:4:[s:4:`type`;s:50:`Magento|CatalogWidget|Model|Rule|Condition|Combine`;s:10:`aggregator`;s:3:`all`;s:5:`value`;s:1:`1`;s:9:`new_child`;s:0:``;]s:4:`1--1`;a:4:[s:4:`type`;s:50:`Magento|CatalogWidget|Model|Rule|Condition|Product`;s:9:`attribute`;s:3:`sku`;s:8:`operator`;s:2:`()`;s:5:`value`;s:63:`24-MB04, 24-MB01, 24-MB03, 24-MB06, 24-MB05, 24-MB02, 24-UB02, 24-WB01, 24-WB02`;]]"}}```
2. Save the CMS block
3. View the Contact Us page
    1. The product list should display as expected 
3. Upgrade to 2.2.0-RC2.1
4. View the Contact Us page

### Expected result
1. Product list would render on the contact us page after upgrading to 2.2.0-RC2.1
2. The widget within the contact-us-info CMS block would have been migrated to JSON data.

### Actual result
1. The contact us page displays an error: "We're sorry, an error has occurred while generating this content."
2. The widget within the contact-us-info CMS block still contains seralized data

**Comment by misha-kotov** [comment]
Thanks for reporting - internal issue is MAGETWO-72233.

**Comment by sidolov** [comment]
Hi @toddwolaver, did you run bin/magento setup:upgrade command after upgrade to 2.2.0-RC2.1 ?

**Comment by magento-engcom-team** [comment]
@toddwolaver, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.

**Comment by andrewhowdencom** [comment]
Please reopen. cc @orlangur as you were so patient with the last issue, and are probably still about. <3

Also hitting this.

**Comment by orlangur** [comment]
@andrewhowdencom issue seems to be irreproducible with the steps provided. Are able to reproduce it with

1) Install clean 2.1.8 with sample data
2) Create widget as described
3) Upgrade to 2.2 stable

?

**Comment by andrewhowdencom** [comment]
I haven't tried (and don't have the time budget right now); I've got a store in an existing state with a large number of custom widgets, but that is using core code. While it might be difficult to reproduce, there are now at least two instances which have occurred, and I think it should remain open.

I can repro it in this case easily -- the work currently is to upgrade this store.

Additionally there's nothing that would migrate this data in the CMS module -- ipso facto, it might be broken and we should leave it discoverable

oh wait found it. leave this with me I might be nuts. 

Investigating: 

```
        if (version_compare($context->getVersion(), '2.0.2', '<')) {
            $this->convertWidgetConditionsToJson($setup);
        }
```

Ref: https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Cms/Setup/UpgradeData.php#L79-L81

Only in 2.2.  https://github.com/magento/magento2/commit/4c44d3cb1cfc8b00fd098826ea110f1dae7b4a34

2.0.2 is module version. lol

conditions vs conditions encoded?

**Comment by orlangur** [comment]
`2.0.2` should be fine here: https://github.com/magento/magento2/blob/2.1/app/code/Magento/Cms/etc/module.xml#L9
https://github.com/magento/magento2/blob/2.2/app/code/Magento/Cms/etc/module.xml#L9

`upgradeVersionTwoZeroTwo` was an interesting name choice :) Surely, there is no need in clean try if there is some broken store in your hand. Can you confirm that this upgrade was executed successfully but some of the widgets remain unconverted? Maybe it does not work properly in some corner case. It would be super helpful if you provide exact condition in such widget then.

**Comment by andrewhowdencom** [comment]
I am super embarrassed to say it looks like third party code that overloads the widget module with an additional field which is, in turn, encoded. :disappointed: I am really the boy who cried wolf today. If it turns out otherwise, I will let you know.

I was so excited! I love core bugs lol.

**Comment by orlangur** [comment]
Ok 👍 Actually, all extensions should have a corresponding `< 2.2` constraint until they are fixed for `2.2`. https://devdocs.magento.com/guides/v2.2/release-notes/backward-incompatible-changes.html#database-data-format-changes describes action points for achieving this compatibility.
